Assets/MixedRealityToolkit-SDK/Features/Boundary/System/MixedRealityBoundarySystem.cs

```diff
@@ -0,0 +1,491 @@
+        }
+
+        /// <inheritdoc />
+        public int GetHashCode(object obj)
+        {
+            return Mathf.Abs(SourceName.GetHashCode());
+        }
+
+        /// <inheritdoc />
+        public uint SourceId { get; } = 0;
+
+        /// <inheritdoc />
+        public string SourceName { get; } = "Mixed Reality Boundary System";
+
+        #endregion IMixedRealityEventSource Implementation
+
+        #region IMixedRealityBoundarySystem Implementation
+
+        /// <inheritdoc/>
+        public ExperienceScale Scale { get; set; }
+
+        /// <inheritdoc/>
+        public float BoundaryHeight { get; set; } = 3f;
+
+        // todo: coming in Beta
+        //// Default to not showing the boundary walls.
+        // private bool showBoundaryWallss = false;
+        ///// <inheritdoc/>
+        //public bool ShowBoundaryWalls { get; set; } = false;
+
+        // todo: coming in Beta
+        //// Default to not showing the ceiling plane.
+        // private bool showCeiling = false;
+        ///// <inheritdoc/>
+        //public bool ShowCeiling { get; set; } = false;
+
+        // Default to showing the floor plane.
+        private bool showFloor = true;
+
+        /// <inheritdoc/>
+        public bool ShowFloor
+        {
+            get { return showFloor; }
+            set
+            {
+                if (showFloor != value)
+                {
+                    showFloor = value;
+
+                    if (value && (currentFloorPlane == null))
+                    {
+                        GetFloorVisualization();
+                    }
+
+                    currentFloorPlane?.SetActive(value);
+
+                    RaiseBoundaryVisualizationChanged();
+                }
+            }
+        }
+
+        private bool showPlayArea = true;
+
+        /// <inheritdoc/>
+        public bool ShowPlayArea
+        {
+            get { return showPlayArea; }
+            set
+            {
+                if (showPlayArea != value)
+                {
+                    showPlayArea = value;
+
+                    if (value && (currentPlayArea == null))
+                    {
+                        GetPlayAreaVisualization();
+                    }
+
+                    currentPlayArea?.SetActive(value);
+
+                    RaiseBoundaryVisualizationChanged();
+                }
+            }
+        }
+
+        // todo: coming in Beta
+        //// Default to not showing the tracked area.
+        // private bool showTrackedArea = false;
+        ///// <inheritdoc/>
+        //public bool ShowTrackedArea { get; set; } = false;
+
+        /// <inheritdoc/>
+        public Edge[] Bounds { get; private set; } = new Edge[0];
+
+        /// <inheritdoc/>
+        public float? FloorHeight { get; private set; } = null;
+
+        /// <inheritdoc/>
+        public bool Contains(Vector3 location, UnityEngine.Experimental.XR.Boundary.Type boundaryType = UnityEngine.Experimental.XR.Boundary.Type.TrackedArea)
```

Comment:
since we've declaired the using at the top do we need ` UnityEngine.Experimental.XR` here?

Reply:
nope. missed removing that one. nice catch